### ISSUE TITLE
tsc error: Could not find a declaration file for module 'nsql-cache'

### ISSUE THREAD

**Issue by reedom** [sebelga/gstore-node#223]
When I invoke `tsc` upon my project, I got the following error in `node_modules/gstore-node/lib/index`, due to lack of `nsql-cache` module's type definition.
How other people overcome this?

```
 error TS7016: Could not find a declaration file for module 'nsql-cache'. '/path/to/myproj/node_modules/nsql-cache/index.js' implicitly has an 'any' type.
  Try `npm install @types/nsql-cache` if it exists or add a new declaration (.d.ts) file containing `declare module 'nsql-cache';`

1 import NsqlCache, { NsqlCacheConfig } from 'nsql-cache';
                                             ~~~~~~~~~~~~
```

As workarounds, I could take the ways below, but I feel like none of them are best.
a. turn off `strict` flag in `tsconfig.json`
b. include [`sql-cache`'s type definition in `gstore-node` repository][] somehow

[`sql-cache`'s type definition in `gstore-node` repository]: https://github.com/sebelga/gstore-node/blob/master/typings/nsql-cache-datastore/index.d.ts

**Comment by sebelga** [comment]
Hi @reedom, sorry for the late answer!

Have you tried copying the declaration in your project from the repo: https://github.com/sebelga/gstore-node/blob/master/typings/nsql-cache/index.d.ts

Cheers! 👍

**Comment by reedom** [comment]
Yes, I had tried and then posted this issue, while wondering it was whether the expected solution or just a workaround, or even it was only me who face this kind of type-not-found error.

And my question hasn't changed ever since.

**Comment by sebelga** [comment]
Hello,

So indeed you need to copy the `nsql-cache` typing from this repo into yours.

I added a `"typings"` folder at the root with this folder structure

```
- typings
  - nsql-cache
    - index.d.ts
```

Then in you `tsconfig.json` file you need to add this folder

```
// tsconfig.json
{
  "compilerOptions": {
    ...
  },
  "include": [
    "src",
    "typings"
  ]
}
```

Let me know if it works for you.

**Comment by carnun** [comment]
Closing the issue since there has been no response.
